store/driver/txn/ranged_kv_retriever.go

```diff
@@ -62,3 +65,158 @@ func (s *RangedKVRetriever) Intersect(startKey, endKey kv.Key) *RangedKVRetrieve
 
 	return nil
 }
+
+// ScanCurrentRange scans the retriever for current range
+func (s *RangedKVRetriever) ScanCurrentRange(reverse bool) (kv.Iterator, error) {
+	if !s.Valid() {
+		return nil, errors.New("retriever is invalid")
+	}
+
+	startKey, endKey := s.StartKey, s.EndKey
+	if !reverse {
+		return s.Iter(startKey, endKey)
+	}
+
+	iter, err := s.IterReverse(endKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(startKey) > 0 {
+		iter = newLowerBoundReverseIter(iter, startKey)
+	}
+
+	return iter, nil
+}
+
+type sortedRetrievers []*RangedKVRetriever
+
+func (retrievers sortedRetrievers) TryGet(ctx context.Context, k kv.Key) (bool, []byte, error) {
+	if len(retrievers) == 0 {
+		return false, nil, nil
+	}
+
+	_, r := retrievers.searchRetrieverByKey(k)
+	if r == nil || !r.Contains(k) {
+		return false, nil, nil
+	}
+
+	val, err := r.Get(ctx, k)
+	return true, val, err
+}
+
+func (retrievers sortedRetrievers) TryBatchGet(ctx context.Context, keys []kv.Key, collectF func(k kv.Key, v []byte)) ([]kv.Key, error) {
+	if len(retrievers) == 0 {
+		return keys, nil
+	}
+
+	// assign keys to nonCustomKeys, if all the keys are non-custom, no new space will be allocated
+	nonCustomKeys := keys
+	hasCustomKeys := false
```

Comment:
It seems that `hasCustomKeys` is always the same as `nonCustomKeys != keys`. So how about remove the flag and replace `if hasCustomKeys` with `if nonCustomKeys != keys` and `if !hasCustomKeys` with `if nonCustomKeys == keys`?

Reply:
golang do not allow "==" between slices.